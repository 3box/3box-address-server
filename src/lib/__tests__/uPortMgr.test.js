const UportMgr = require("../uPortMgr");

describe("UportMgr", () => {
  let sut;
  let jwts = {
    legacy:
      "eyJ0eXAiOiJKV1QiLCJhbGciOiJFUzI1NksifQ.eyJpc3MiOiIyb3pzRlFXQVU3Q3BIWkxxdTJ3U1liSkZXekROQjI2YW9DRiIsImlhdCI6MTUxMzM1MjgzNCwiZXZlbnQiOnsidHlwZSI6IlNUT1JFX0NPTk5FQ1RJT04iLCJhZGRyZXNzIjoiMm96c0ZRV0FVN0NwSFpMcXUyd1NZYkpGV3pETkIyNmFvQ0YiLCJjb25uZWN0aW9uVHlwZSI6ImNvbnRyYWN0cyIsImNvbm5lY3Rpb24iOiIweDJjYzMxOTEyYjJiMGYzMDc1YTg3YjM2NDA5MjNkNDVhMjZjZWYzZWUifSwiZXhwIjoxNTEzNDM5MjM0fQ.tqX5eEuaTEyYPUSgatK5zEsj_WpE-dIEHDc4ItpOvAZuBkSyV9_zbb0puNtDrZTVA7MlZ43FSSpf9CGIUxup-w"
  };
  beforeAll(() => {
    sut = new UportMgr();
    jasmine.DEFAULT_TIMEOUT_INTERVAL = 60000;
  });

  test("empty constructor", () => {
    expect(sut).not.toBeUndefined();
  });

  test("verifyToken() no token", done => {
    sut
      .verifyToken(null)
      .then(resp => {
        fail("shouldn't return");
        done();
      })
      .catch(err => {
        expect(err.message).toEqual("no token");
        done();
      });
  });

  // This test isn't working, using uport did resolver
  // We need to update it to use muport (and 3id in future)
  describe.skip("verifyToken() happy path", () => {
    const DATE_TO_USE = new Date(1513399280000);
    Object.keys(jwts).forEach(didType => {
      test(didType, done => {
        const eventToken = jwts[didType];
        Date.now = jest.fn(() => DATE_TO_USE);

        sut.verifyToken(eventToken).then(
          resp => {
            expect(resp.jwt).toEqual(eventToken);
            expect(resp.payload.event.address).toEqual(
              "2ozsFQWAU7CpHZLqu2wSYbJFWzDNB26aoCF"
            );
            done();
          },
          error => {
            expect(error).toBeUndefined();
            done();
          }
        );
      });
    });
  });
});
