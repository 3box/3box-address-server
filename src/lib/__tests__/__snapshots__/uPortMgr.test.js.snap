// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UportMgr should verify 3id JWT correctly 1`] = `
Object {
  "doc": Object {
    "@context": "https://w3id.org/did/v1",
    "authentication": Array [
      Object {
        "publicKey": "did:3:bafyreifhpmpwpwmjhd64vcsc35fsgjgpdrqqdnsyjv45ggbzn3l5m4feda#signingKey",
        "type": "Secp256k1SignatureAuthentication2018",
      },
    ],
    "id": "did:3:bafyreifhpmpwpwmjhd64vcsc35fsgjgpdrqqdnsyjv45ggbzn3l5m4feda",
    "publicKey": Array [
      Object {
        "id": "did:3:bafyreifhpmpwpwmjhd64vcsc35fsgjgpdrqqdnsyjv45ggbzn3l5m4feda#signingKey",
        "publicKeyHex": "04113970c9f2818bf183ee0cce54446fce4f6e1303d5b0d380ec95b5cf97e2f866403a204a0fda234f132a437ee0c56bd2a44dff661d8546dcb4132930ed36907a",
        "type": "Secp256k1VerificationKey2018",
      },
      Object {
        "id": "did:3:bafyreifhpmpwpwmjhd64vcsc35fsgjgpdrqqdnsyjv45ggbzn3l5m4feda#encryptionKey",
        "publicKeyBase64": "somakoiV8ZenByzOhk2Z6jxM2WCHtPB6KoPKj26YMFM=",
        "type": "Curve25519EncryptionPublicKey",
      },
      Object {
        "ethereumAddress": "0x456bf7002f2377798ef546fb7a0D0eE6155E02e5",
        "id": "did:3:bafyreifhpmpwpwmjhd64vcsc35fsgjgpdrqqdnsyjv45ggbzn3l5m4feda#managementKey",
        "type": "Secp256k1VerificationKey2018",
      },
    ],
  },
  "issuer": "did:3:bafyreifhpmpwpwmjhd64vcsc35fsgjgpdrqqdnsyjv45ggbzn3l5m4feda",
  "jwt": "eyJ0eXAiOiJKV1QiLCJhbGciOiJFUzI1NksifQ.eyJpYXQiOjE1NzQ2NzIyNTQsInJvb3RTdG9yZUFkZHJlc3MiOiIvb3JiaXRkYi9RbVNQS2lGaWQxdGRKWlhKcEptY2RNYm9ZakhiaDM0VGM5cTM4VTJvNlc4Z0VDLzEyMjBkNjE2NjljNzU5MGQ1MjI2NDU1MjIzNGMwOWM3ZmRmZTM3MTlmNDg3NjhkMGU0ZmNiNzA2YmJhYTg4YzQ4OGI0LnJvb3QiLCJpc3MiOiJkaWQ6MzpiYWZ5cmVpZmhwbXB3cHdtamhkNjR2Y3NjMzVmc2dqZ3BkcnFxZG5zeWp2NDVnZ2J6bjNsNW00ZmVkYSJ9.PG4U6Awwu-VCMDS12hnGeJ1Zop2cyL-LbXJnZGVgYsxZzTCmfrz-uNlMbRZ2r0al4WQg-dMGN6FgSNWmqEIPMQ",
  "payload": Object {
    "iat": 1574672254,
    "iss": "did:3:bafyreifhpmpwpwmjhd64vcsc35fsgjgpdrqqdnsyjv45ggbzn3l5m4feda",
    "rootStoreAddress": "/orbitdb/QmSPKiFid1tdJZXJpJmcdMboYjHbh34Tc9q38U2o6W8gEC/1220d61669c7590d52264552234c09c7fdfe3719f48768d0e4fcb706bbaa88c488b4.root",
  },
  "signer": Object {
    "id": "did:3:bafyreifhpmpwpwmjhd64vcsc35fsgjgpdrqqdnsyjv45ggbzn3l5m4feda#signingKey",
    "publicKeyHex": "04113970c9f2818bf183ee0cce54446fce4f6e1303d5b0d380ec95b5cf97e2f866403a204a0fda234f132a437ee0c56bd2a44dff661d8546dcb4132930ed36907a",
    "type": "Secp256k1VerificationKey2018",
  },
}
`;
